Framework/Kernel/test/RebinHistogramTest.h

```diff
@@ -52,4 +54,77 @@ class RebinHistogramTest : public CxxTest::TestSuite {
   }
 };
 
+class RebinHistogramTestPerformance : public CxxTest::TestSuite {
+public:
+  RebinHistogramTestPerformance *createSuite() {
+    return new RebinHistogramTestPerformance();
+  }
+
+  void destroySuite(RebinHistogramTestPerformance *suite) { delete suite; }
+
+  RebinHistogramTestPerformance() {
+    setupHistogram();
+    setupOutput();
+  }
+
+  void testRebinSmaller() {
+    auto size = smallerBinEdges.size() - 1;
+    for (size_t i = 0; i < nIters; i++) {
+      std::vector<double> yout(size);
+      std::vector<double> eout(size);
+      Mantid::Kernel::VectorHelper::rebinHistogram(
+          binEdges, counts, errors, smallerBinEdges, yout, eout, false);
+    }
+  }
+
+  void testRebinLarger() {
+    auto size = largerBinEdges.size() - 1;
+    for (size_t i = 0; i < nIters; i++) {
+      std::vector<double> yout(size);
+      std::vector<double> eout(size);
+      Mantid::Kernel::VectorHelper::rebinHistogram(
+          binEdges, counts, errors, largerBinEdges, yout, eout, false);
+    }
+  }
+
+private:
+  const size_t binSize = 10000;
+  const size_t nIters = 10000;
+  std::vector<double> binEdges;
+  std::vector<double> counts;
+  std::vector<double> errors;
+  std::vector<double> smallerBinEdges;
+  std::vector<double> largerBinEdges;
+
+  void setupHistogram() {
+    binEdges.resize(binSize);
+    counts.resize(binSize - 1);
+    errors.resize(binSize - 1);
+
+    std::iota(binEdges.begin(), binEdges.end(), 0);
+    std::generate(counts.begin(), counts.end(),
+                  []() { return static_cast<double>(rand() % 1000); });
+
+    std::transform(counts.cbegin(), counts.cend(), errors.begin(),
+                   [](const double count) { return sqrt(count); });
+  }
```

Comment:
Why all the effort to make this tests data? Should not matter for a performance test.